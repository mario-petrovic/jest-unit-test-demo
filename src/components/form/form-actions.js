export const ActionTypes = {
    SET_NAME: 'SET_NAME',
    GET_NAME: 'GET_NAME'
};


export /* istanbul ignore next */ function setName(dispatch, name) {
    /* istanbul ignore next */
    dispatch({
        type: ActionTypes.SET_NAME,
        name
    });
}


export /* istanbul ignore next */ function getName() {
    /* istanbul ignore next */
    return{
        type: ActionTypes.GET_NAME,
        name
    };
}