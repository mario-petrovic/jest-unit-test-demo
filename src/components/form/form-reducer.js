import { ActionTypes } from './form-actions';

export const INITIAL_STATE = {
    name: '',
    surname: ''
};

export default function formReducer (state = INITIAL_STATE, action) {
    switch (action.type) {
        case ActionTypes.SET_NAME:
            return {
                ...state,
                name: action.name
            };

        default:
            
            return state;
    }
};