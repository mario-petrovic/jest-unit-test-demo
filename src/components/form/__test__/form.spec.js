import React from 'react'
import { shallow, mount } from 'enzyme';
import ConnectedForm, { Form } from '../form';
import configureStore from 'redux-mock-store'
import { INITIAL_STATE } from '../form-reducer';
import { setName, getName } from '../form-actions';
import { ActionTypes } from './../form-actions';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

describe('>>>Form component', () => {
    let wrapper;
    const initialState = {
        name: '',
        surname: ''
    };
    const mockStore = configureStore()
    var container;
    var store;

    beforeEach(() => {
        store = mockStore(initialState)
        wrapper = mount(<Provider store={store}><ConnectedForm /></Provider>);
    })

    it('+++ render Form component', () => {
        expect(wrapper.find(ConnectedForm).length).toEqual(1);
    });

    it('+++ check if props match with initial store', () => {
        expect(wrapper.find(Form).prop('name')).toEqual(initialState.name);
    });

    it('+++ check action on dispatching ', () => {
        let action
        store.dispatch(getName('macika'));
        action = store.getActions()
        expect(action[0].type).toBe(ActionTypes.GET_NAME)
    });
});


// Snapshot for Home React Component
describe('>>>H O M E --- Snapshot',()=>{
    it('+++capturing Snapshot of Home', () => {
        const renderedValue =  renderer.create(<Form inputValue={10}/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});