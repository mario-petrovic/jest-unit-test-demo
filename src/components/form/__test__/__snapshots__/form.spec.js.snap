// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>H O M E --- Snapshot +++capturing Snapshot of Home 1`] = `
<div>
  <form>
    <input
      type="text"
      value={10}
    />
  </form>
</div>
`;
